src/common/tests/umem_test_bmem.c

```diff
@@ -830,6 +881,41 @@ test_alloc(void **state)
 	assert_int_equal(rc, 0);
 }
 
+static void
+test_alloc_from_bucket(void **state)
+{
+	struct test_arg		*arg = *state;
+	struct umem_instance	*umm = utest_utx2umm(arg->ta_utx);
+	int			*value1;
+	umem_off_t		 umoff = 0;
+	int			 rc;
+
+	rc = utest_tx_begin(arg->ta_utx);
+	if (rc != 0)
+		goto done;
+
+	umoff = umem_zalloc_from_bucket(umm, 4, UMEM_DEFAULT_MBKT_ID);
+	if (UMOFF_IS_NULL(umoff)) {
+		print_message("umoff unexpectedly NULL\n");
+		rc = 1;
+		goto end;
+	}
```

Comment:
Why use two different error-handling methods?
Above you used asserts e.g. `assert_true(UMOFF_IS_NULL(off));`.

Reply:
I have addressed this.  I am now required to move this PR to new branch. Pls review the rework in https://github.com/daos-stack/daos/pull/13152